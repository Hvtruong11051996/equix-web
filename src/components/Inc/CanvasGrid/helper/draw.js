import { path } from '../../SvgIcon';

export function drawPath2D(params, svgPath, x, y, w, h) {
    const { gc } = params;
    const p = new Path2D();
    const min = Math.min(w / (svgPath.w || 24), h / (svgPath.h || 24));
    p.addPath(new Path2D(svgPath.d), {
        a: min,
        d: min,
        e: x + 0.5,
        f: y + 0.5
    });
    gc.fill(p);
}
export function drawProgressAlert(params, x, y, w, h) {
    drawPath2D(params, path.mdiProgress, x, y, w, h);
}
export function drawProgressPending(params, x, y, w, h) {
    drawPath2D(params, path.progressPending, x, y, w, h);
}
export function drawLockout(params, x, y, w, h) {
    drawPath2D(params, path.mdiLock, x, y, w, h);
}
export function drawCalendar(params, x, y, w, h) {
    drawPath2D(params, path.mdiCalendarBlank, x, y, w, h);
}
export function drawSwitchOff(params, x, y, w, h) {
    drawPath2D(params, path.mdiToggleSwitchOff, x, y, w, h);
}
export function drawSwitch(params, x, y, w, h) {
    drawPath2D(params, path.mdiToggleSwitch, x, y, w, h);
}
export function drawSearch(params, x, y, w, h) {
    drawPath2D(params, path.mdiMagnify, x, y, w, h);
}
export function drawDownload(params, x, y, w, h) {
    drawPath2D(params, path.mdiDownload, x, y, w, h);
}
export function drawSquareCheckbox(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheckBoxOutline, x, y, w, h);
}
export function drawUncheckSquare(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheckboxBlankOutline, x, y, w, h);
}
export function drawActiveIcon(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheckCircle, x, y, w, h);
}
export function drawCloseIcon(params, x, y, w, h) {
    drawPath2D(params, path.mdiCloseCircle, x, y, w, h);
}
export function drawPendingIcon(params, x, y, w, h) {
    drawPath2D(params, path.mdiRestart, x, y, w, h);
}
export function drawMenuIcon(params, x, y, w, h) {
    drawPath2D(params, path.mdiMenu, x, y, w, h);
}
export function drawFilter(params, x, y, w, h) {
    drawPath2D(params, path.mdiFilter, x, y, w, h)
}
export function drawCheck(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheck, x, y, w, h)
}
export function drawCheckboxIntermediate(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheckboxIntermediate, x, y, w, h)
}
export function drawCheckbox(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheckboxMarkedOutline, x, y, w, h)
}
export function drawCheckboxEmpty(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheckboxBlankOutline, x, y, w, h)
}
export function drawCheckCircle(params, x, y, w, h) {
    drawPath2D(params, path.mdiCheckboxMarkedCircle, x, y, w, h)
}
export function drawCloseCircle(params, x, y, w, h) {
    drawPath2D(params, path.mdiCloseCircle, x, y, w, h)
}
export function drawCheckPending(params, x, y, w, h) {
    drawPath2D(params, path.checkPending, x, y, w, h)
}
export function drawSortAsc(params, x, y, w, h) {
    drawPath2D(params, path.mdiSortAlphabeticalAscending, x, y, w, h)
}
export function drawSortDesc(params, x, y, w, h) {
    drawPath2D(params, path.mdiSortAlphabeticalDescending, x, y, w, h)
}
export function drawFlag(params, x, y, w, h) {
    drawPath2D(params, path.mdiFlag, x, y, w, h)
}
export function drawArrowUp(params, x, y, w, h) {
    drawPath2D(params, path.mdiChevronUp, x, y, w, h)
}
export function drawArrowDown(params, x, y, w, h) {
    drawPath2D(params, path.mdiChevronDown, x, y, w, h)
}
export function drawArrowRight(params, x, y, w, h) {
    drawPath2D(params, path.mdiChevronRight, x, y, w, h)
}
export function drawCartPlus(params, x, y, w, h) {
    drawPath2D(params, path.mdiCartPlus, x, y, w, h)
}
export function drawCartOff(params, x, y, w, h) {
    drawPath2D(params, path.mdiCartOff, x, y, w, h)
}
export function drawClose(params, x, y, w, h) {
    drawPath2D(params, path.mdiClose, x, y, w, h)
}
export function drawContrast(params, x, y, w, h) {
    drawPath2D(params, path.mdiContrast, x, y, w, h)
}
export function drawInformationOutline(params, x, y, w, h) {
    drawPath2D(params, path.mdiInformationOutline, x, y, w, h)
}
export function drawPlayListRemove(params, x, y, w, h) {
    drawPath2D(params, path.mdiPLaylistRemove, x, y, w, h)
}
export function drawDragBtn(params, x, y, w, h) {
    drawPath2D(params, path.dragBtn, x, y, w, h)
}
export function drawModifyBtn(params, x, y, w, h) {
    drawPath2D(params, path.mdiFileDocumentEditOutline, x, y, w, h)
}
export function drawLoading(params, x, y, w, h) {
    drawPath2D(params, path.loading, x, y, w, h)
}
export function drawCreatingAccount(params, x, y, w, h) {
    drawPath2D(params, path.creatingAccount, x, y, w, h)
}
export function drawEkycInProgress(params, x, y, w, h) {
    drawPath2D(params, path.ekycInProgress, x, y, w, h)
}
export function drawMdiBellIcon(params, x, y, w, h) {
    drawPath2D(params, path.mdiBellOutLine, x, y, w, h)
}
export function drawSmsIcon(params, x, y, w, h) {
    drawPath2D(params, path.smsIcon, x, y, w, h)
}
export function drawEmailIcon(params, x, y, w, h) {
    drawPath2D(params, path.mdiEmailOutline, x, y, w, h)
}

export function drawPalette(params, x, y, w, h) {
    drawPath2D(params, path.mdiPaletteOutline, x, y, w, h)
}

export function drawPen(params, x, y, w, h) {
    drawPath2D(params, path.mdiPen, x, y, w, h)
}

export function drawLoad(params, x, y, w, h) {
    drawPath2D(params, path.mdiLoad, x, y, w, h)
}
