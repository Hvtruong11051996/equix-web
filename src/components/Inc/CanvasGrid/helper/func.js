
export const calculateWidth = type => {
    return (function () {
        try {
            return require('../Type/' + type + '.js').size;
        } catch (ex) {
            return require('../Type/label.js').size
        }
    })()
}
export function truncateText(params, text = '', maxWidth) {
    const { gc } = params;
    let width = gc.measureText(text).width;
    const ellipsis = 'â€¦';
    const ellipsisWidth = gc.measureText(ellipsis).width;
    if (width <= maxWidth || width <= ellipsisWidth) {
        return text;
    } else {
        var len = text.length;
        while (width >= maxWidth - ellipsisWidth && len-- > 0) {
            text = text.substring(0, len);
            width = gc.measureText(text).width;
        }
        return text + ellipsis;
    }
}
export function pointInPath(point, objPoint) {
    return point.x >= objPoint.x && point.x <= objPoint.x + objPoint.w && point.y >= objPoint.y && point.y <= objPoint.y + objPoint.h
}
export function showTooltip(textTran, x, y, style) {
    if (!textTran) return;
    let div = document.getElementById('tooltip');
    if (!div) {
        div = document.createElement('div');
        div.id = 'tooltip';
        div.style.position = 'absolute';
        div.classList.add('size--2');
        document.body.appendChild(div);
    }
    div.innerHTML = textTran;
    if (style) Object.assign(div.style, style);
    const rect = div.getBoundingClientRect();
    const windowHeight = document.body.clientHeight;
    const windowWidth = document.body.clientWidth;
    if (x + 10 + rect.width >= windowWidth) {
        div.style.left = windowWidth - rect.width + 'px';
    } else {
        div.style.left = x + 10 + 'px';
    }
    if (y + 10 + rect.height >= windowHeight) {
        div.style.top = windowHeight - rect.height + 'px';
    } else {
        div.style.top = y + 10 + 'px';
    }
    div.style.opacity = 1;
}
export function hideTooltip(style) {
    const div = document.getElementById('tooltip');
    if (div) {
        div.style.opacity = 0;
        if (style) Object.assign(div.style, style);
    }
}
