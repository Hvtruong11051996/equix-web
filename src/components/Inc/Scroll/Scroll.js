import scrollbarWidth from 'scrollbarwidth';

export default class Scroll {
  constructor(scroll) {
    if (!scroll || scroll.hasAttribute('scroll-content')) return;
    scroll.setAttribute('scroll-content', '');
    scroll.style.bottom = scroll.style.right = (0 - scrollbarWidth()) + 'px';
    const container = scroll.parentNode;
    container.setAttribute('scroll', '');
    let div;
    let bar1;
    scroll.childNodes.forEach(node => {
      if (node.classList && node.classList.contains('vertical')) {
        bar1 = node.firstChild
      }
    })
    if (!bar1) {
      div = document.createElement('div');
      div.className = 'vertical';
      div.style.display = 'none';
      bar1 = document.createElement('div');
      bar1.className = 'scrollBar';
      div.appendChild(bar1);
      container.appendChild(div);
    }
    bar1.style.height = (scroll.clientHeight / (scroll.scrollHeight / scroll.clientHeight)) + 'px';
    bar1.onmousedown = function(e) {
      // bar1.dragging = true;
      e = e || window.event;
      // get the mouse cursor position at startup:
      bar1.topPos = e.clientY - bar1.offsetTop;
      document.body.classList.add('noSelect');
      document.onmouseup = function() {
        /* stop moving when mouse button is released: */
        // bar1.dragging = false;
        document.onmouseup = null;
        document.onmousemove = null;
        document.body.classList.remove('noSelect');
      };
      // call a function whenever the cursor moves:
      document.onmousemove = function(e) {
        e = e || window.event;
        let top = e.clientY - bar1.topPos;

        if (top < 0) top = 0;
        else if (top > container.clientHeight - bar1.clientHeight) top = container.clientHeight - bar1.clientHeight;
        // bar1.style.top = top + "px";
        scroll.scrollTop = top * (scroll.scrollHeight - scroll.clientHeight) / (bar1.parentNode.clientHeight - bar1.clientHeight);
        // const elmnt = e.target.parentNode.parentNode;
        // elmnt.style.top = (elmnt.offsetTop - e.target.pos2) + "px";
        // elmnt.style.left = (elmnt.offsetLeft - e.target.pos1) + "px";
      };
    };
    let bar2;
    scroll.childNodes.forEach(node => {
      if (node.classList && node.classList.contains('horizontal')) {
        bar2 = node.firstChild
      }
    })
    if (!bar2) {
      div = document.createElement('div');
      div.className = 'horizontal';
      div.style.display = 'none';
      bar2 = document.createElement('div');
      bar2.className = 'scrollBar';
      div.appendChild(bar2);
      container.appendChild(div);
    }
    bar2.style.width = (scroll.clientWidth / (scroll.scrollWidth / scroll.clientWidth)) + 'px';
    bar2.onmousedown = function(e) {
      // bar1.dragging = true;
      e = e || window.event;
      // get the mouse cursor position at startup:
      bar2.leftPos = e.clientX - bar2.offsetLeft; //
      document.body.classList.add('noSelect');
      document.onmouseup = function() {
        /* stop moving when mouse button is released: */
        // bar1.dragging = false;
        document.onmouseup = null;
        document.onmousemove = null;
        document.body.classList.remove('noSelect');
      };
      // call a function whenever the cursor moves:
      document.onmousemove = function(e) {
        e = e || window.event;
        let left = e.clientX - bar2.leftPos;

        if (left < 0) left = 0;
        else if (left > container.clientWidth - bar2.clientWidth) left = container.clientWidth - bar2.clientWidth;
        scroll.scrollLeft = left * (scroll.scrollWidth - scroll.clientWidth) / (bar2.parentNode.clientWidth - bar2.clientWidth);
      };
    };

    scroll.onscroll = function() {
      if (scroll.onmousemove) scroll.onmousemove();
      bar1.style.transform = `translate(0,${(scroll.scrollTop * (bar1.parentNode.clientHeight - bar1.clientHeight) / (scroll.scrollHeight - scroll.clientHeight))}px)`;
      bar2.style.transform = `translate(${(scroll.scrollLeft * (bar2.parentNode.clientWidth - bar2.clientWidth) / (scroll.scrollWidth - scroll.clientWidth))}px,0)`;
    };
    bar1.parentNode.onmouseenter = bar2.parentNode.onmouseenter = function() {
      scroll.onmousemove = null;
      if (scroll.time) {
        clearTimeout(scroll.time);
      }
      container.setAttribute('active', '');
    };
    bar1.parentNode.onmouseleave = bar2.parentNode.onmouseleave = function() {
      scroll.onmousemove = scroll.timmeout;
      container.removeAttribute('active');
    };
    scroll.timmeout = function() {
      container.setAttribute('active', '');
      if (this.time) {
        clearTimeout(this.time);
      }
      this.time = setTimeout(function() {
        container.removeAttribute('active');
        this.time = null;
      }, 1000);
    };
    scroll.onmousemove = scroll.timmeout;
    scroll.monitor = () => {
      let size = 0;
      if (scroll.lastChild) {
        size = scroll.lastChild.offsetHeight + scroll.lastChild.offsetTop + 1;
        if (scroll.lastElementChild) size += parseInt(getComputedStyle(scroll.lastElementChild).marginBottom);
      }
      if (scroll.parentNode) {
        scroll.parentNode.style.height = size + 'px';
      }
      if (bar1.parentNode) {
        if (scroll.scrollHeight <= scroll.clientHeight) {
          bar1.parentNode.style.display = 'none';
        } else {
          bar1.parentNode.style.display = '';
          size = (scroll.clientHeight / (scroll.scrollHeight / scroll.clientHeight));
          bar1.style.height = (size < 32 ? 32 : size) + 'px';
          bar1.style.transform = `translate(0,${(scroll.scrollTop * (bar1.parentNode.clientHeight - bar1.clientHeight) / (scroll.scrollHeight - scroll.clientHeight))}px)`;
        }
      }
      if (bar2.parentNode) {
        if (scroll.scrollWidth === scroll.clientWidth) {
          bar2.parentNode.style.display = 'none';
        } else {
          bar2.parentNode.style.display = '';
          size = (scroll.clientWidth / (scroll.scrollWidth / scroll.clientWidth));
          bar2.style.width = (size < 32 ? 32 : size) + 'px';
          bar2.style.transform = `translate(${(scroll.scrollLeft * (bar2.parentNode.clientWidth - bar2.clientWidth) / (scroll.scrollWidth - scroll.clientWidth))}px,0)`;
        }
      }
    };
    scroll.addEventListener('animationstart', (event) => {
      if (event && event.animationName !== 'nodeVisible') return;
      const scrollContent = scroll.parentNode.querySelector('[scroll-content]')
      if (scrollContent && scroll.parentNode.classList.contains('ag-body-viewport-wrapper')) {
        const headerContainer = scroll.parentNode.parentNode.parentNode.querySelector('.ag-header-container')
        if (headerContainer) {
          headerContainer.style.left = '0px'
        }
      }
      scroll.monitor()
    }, false);
    container.ro = new MutationObserver(scroll.monitor);
    scroll.monitor();
    setTimeout(() => {
      scroll.monitor();
    }, 500)
    container.ro.observe(scroll, {
      attributes: true,
      childList: true,
      subtree: true
    });
  }
}
