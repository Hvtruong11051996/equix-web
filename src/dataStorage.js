
import requireTime from './constants/require_time';
import config from '../public/config'

const obj = {
    mode: null,
    timeoutNetwork: null,
    isChangePassword: false,
    closeChartLayout: {},
    isSessionAnotherLogin: false,
    dicClearRealtime: {},
    materialFont: null,
    listMapping: [],
    deviceLang: 'en_US',
    onChangeLang: null,
    dicAccounts: {},
    reloadNews: [],
    downloadFunc: {},
    printFunc: {},
    layout_role: null,
    force_update_create_user: false,
    deletingLayout: '',
    deletingChartLayout: '',
    isDragChart: false,
    isDefaultLayout: true,
    chartSelfChange: false,
    logout: null,
    isChangeLayout: false,
    curLayoutVersion: null,
    listLayout: {},
    usingLayout: '',
    curChartLayoutVersion: '1',
    listChartLayout: {},
    usingChartLayout: '',
    listUsingChartLayout: {},
    renderAfterRealtime: null,
    checkVersionLayoutBeforeOverride: null,
    isLogin: false,
    uid: null,
    isGuest: true,
    connected: true,
    full_name: null,
    accessToken: null,
    pin: null,
    user_details: null,
    user_type: 2,
    logId: 0,
    timeInterval: 0,
    isIntervalRealtime: 0,
    deviceId: 0,
    account_id: null,
    accountInfo: null,
    role: null,
    sseRegisters: {},
    handleClickOutside: null,
    typeRegisters: {},
    loginEmail: null,
    accountStreamingUrl: null,
    notify: null,
    userInfo: null,
    accounts: [],
    subPath: '',
    checkUpdate: false,
    checkLoginPin: null,
    intervalRefreshToken: null,
    intervalRefreshTokenGuest: null,
    showNotification: null,
    isShowNotification: true,
    showCancelled: false,
    showExpired: false,
    showFilled: false,
    showOnMarket: false,
    showRejected: false,
    showPartialFill: false,
    showAllNews: true,
    translate: null,
    lang: '',
    dicPersonal: {},
    dicPosition: {},
    dicNotiShowed: {},
    symbolEquity: {},
    clickedMenu: false,
    showPin: false,
    resolution: null,
    dicAccount: {},
    accountsObjDic: {},
    symbolsObjDic: {},
    pinWasChanged: false,
    symbolHistoryInfo: {},
    checkPinCallback: null,
    requireTime: requireTime.ON_CHANGE,
    verifiedPin: false,
    isStayLogin: false,
    getLayout: null,
    listFilledCancelled: {},
    dicInterval: {},
    dicLevel1: {},
    dicDepth: {},
    dicCos: {},
    dicLevel1RealTime: {},
    dicDepthRealTime: {},
    dicCosRealTime: {},
    global_symbol: null,
    listUserWatchlist: {},
    usingWatchlist: '',
    numberNoticeNews: null,
    lastSymbolSearchNews: null,
    removeSaveChartLayout: null,
    usingWatchlist_id: null,
    usingWatchlist_name: null,
    userGroupArray: [],
    userGroupDic: [], // tuanphan use
    dicEmailTemp: [],
    apiAccessDic: [],
    userGroupDicReverse: {},
    dicRole: {},
    marketData: {},
    allowDisableModifyCancelOrder: false,
    callBackReloadTheme: {},
    hrefImg: '',
    color: '#fff',
    linkColor: null,
    lstAccountDropdown: null,
    allowCloseAllWidgets: false,
    timeZone: '',
    symbolDefault: null,
    setVisibility: {},
    dicbyDisplayName: {},
    dicsymbol: {},
    watchlist: [],
    openNameWidget: '',
    symbolObjDepth: {},
    dictAccountNoti: [],
    beforeLogin: false,
    timeHoliday: {},
    isHolidayAU: {},
    isHolidayUS: {},
    timeServer: '',
    currentEnv: '',
    env_config: null,
    web_config: null,
    guest_config: null,
    isWarning: {
        isNotiTechnical: false,
        isNotiCashNegative: false
    },
    warningFunc: null,
    listFunctionInit: [],
    marketDataType: {},
    openingAccount: {},
    branchObjDic: {},
    session: {}
}

const objProxy = new Proxy(obj, {
    set: function (target, key, value) {
        // console.log(`YOLO ${key} ${target[key]} =====> ${value}`)
        target[key] = value;
        return true;
    }
})

export default objProxy
