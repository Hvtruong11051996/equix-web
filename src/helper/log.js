import { exception } from 'react-ga';
import { sendLogToServer } from './functionUtils'
const info = (...arg) => {
    console.info(...arg);
}
const error = (...arg) => {
    console.error(...arg);
}
const warn = (...arg) => {
    sendLogToServer(...arg);
    console.warn(...arg);
}
const log = (...arg) => {
    console.log(...arg);
    sendLogToServer(...arg);
}

const sendLog = (...arg) => {
    sendLogToServer(...arg);
}
const sendLogError = (...arg) => {
    sendLogToServer(...arg, 'error');
}

export default {
    info,
    error,
    warn,
    log,
    sendLogError,
    sendLog
}
