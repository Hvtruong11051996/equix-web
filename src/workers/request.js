export default () => {
    // importScripts('https://cdnjs.cloudflare.com/ajax/libs/axios/0.19.0/axios.min.js')
    self.addEventListener('message', e => { // eslint-disable-line
        // eslint-disable-line no-restricted-globals
        // console.log('call worker request');
        if (!e) return;
        // console.log('call worker: ', e.data);
        const thenFn = (res) => {
            // console.log('call worker res: ', res);
            postMessage(res);
        }
        const catchFn = (res) => {
            // console.log('call worker res catch: ', res);
            postMessage(res);
        }
        // console.log('==========>axios', axios.get)
        const dataRecive = JSON.parse(e.data);
        const { method, url, data, token, headers } = dataRecive;
        // console.log('url: ', url)
        // console.log('token: ', token)
        // debugger; // eslint-disable-line
        const xhr = new XMLHttpRequest();
        xhr.open('GET', url, true);
        xhr.setRequestHeader('Authorization', 'Bearer ' + token);
        xhr.onerror = function (e) {
            catchFn({ error: 'network_error' })
        };
        // Now set response type
        let response = { response: xhr.response }
        xhr.addEventListener('load', () => {
            const type = xhr.getResponseHeader('Content-Type') || ''
            if (xhr.status === 200) {
                thenFn({ status: 200, data: xhr.response, type })
            }
            if (xhr.status >= 400) {
                catchFn({ status: xhr.status, error: response, type })
            }
        })
        xhr.send();
    });
};
