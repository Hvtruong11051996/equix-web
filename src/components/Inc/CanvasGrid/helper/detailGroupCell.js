import { showTooltip, hideTooltip } from '../helper/func';
import { drawArrowDown, drawArrowRight } from '../helper/draw';
export default (params) => {
    const { gc, config, style, value, data, grid, name, setRowDetailExpand, expand, rowHover, subPaint } = params
    const b = config.bounds
    const paddingLeft = params.level * 16;
    const draw = () => {
        gc.beginPath();

        if (params.waiting) {
            gc.fillStyle = params.rowStyle;
            gc.fillRect(b.x, b.y, b.width, b.height);
        }
        if (rowHover) {
            gc.fillStyle = style.getPropertyValue('--ascend-default');
            gc.fillRect(paddingLeft + 4, b.y + 4, 24, 24)
        }
        gc.fillStyle = style.getPropertyValue('--secondary-default');
        if (params.waiting) drawArrowRight(params, paddingLeft + Math.round((params.time / 100) % 8), b.y + 4, 24, 24);
        else if (expand) drawArrowDown(params, paddingLeft + 4, b.y + 4, 24, 24);
        else drawArrowRight(params, paddingLeft + 4, b.y + 4, 24, 24);

        gc.closePath();

        gc.beginPath();
        gc.textAlign = 'left';
        const p = { ...params, colDef: { ...params.colDef, align: 'left' }, config: { ...config, bounds: { x: paddingLeft + 28, y: b.y, width: b.width - paddingLeft - 28, height: b.height } } }
        subPaint(p);
        gc.closePath();
    }
    draw();
    return {
        frame: (time) => {
            if (params.waiting) {
                params.time = time;
                draw();
            }
        },
        click: (e) => {
            if (params.waiting) return;
            const p = e.detail.gridPoint;
            if (p.x >= paddingLeft + 4 && p.x <= paddingLeft + 28 && p.y >= b.y + 4 && p.y <= b.y + 28) {
                e.preventDefault();
                setRowDetailExpand(!expand);
                if (!expand && !Array.isArray(params.root._detail[params.key])) {
                    params.waiting = true;
                    params.root.props.detailSource(data, (detailData) => {
                        params.waiting = false;
                        params.root._detail[params.key] = detailData;
                        params.root.setData();
                    });
                }
            }
        },
        mouseMove: e => {
            const p = e.detail.gridPoint;
            if (p.x >= paddingLeft + 4 && p.x <= paddingLeft + 28 && p.y >= b.y + 4 && p.y <= b.y + 28) {
                grid.div.style.cursor = 'pointer';
            }
            const rect = grid.div.getBoundingClientRect();
            showTooltip(params.displayValue, rect.x + e.detail.gridPoint.x + 10, rect.y + e.detail.gridPoint.y);
        },
        mouseLeave: e => {
            hideTooltip();
        }
    }
}
