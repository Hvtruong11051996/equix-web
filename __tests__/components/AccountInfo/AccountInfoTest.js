import Compoment from '../../../src/components/AccountInfo/AccountInfo';
import * as request from '../../../src/helper/request';
import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react'
import dataStorage from '../../../src/dataStorage'
import log from '../../../src/helper/log';
import * as functionUtils from '../../../src/helper/functionUtils';
import Scroll from '../../../src/components/Inc/Scroll/Scroll';

global.MutationObserver = class {
    disconnect() { }
    observe(element, initObject) { }
};

const props1 = {
    loadState: function() {
        return {
            accountObj: {
                currency: 1,
                account_id: 1,
                account_name: 1,
                hin: 1,
                work_phone: 1,
                home_phone: 1,
                fax: 1,
                email: 1,
                address: 1,
                account_type: 1,
                advisor_code: 1,
                advisor_name: 1
            }
        }
    },
    glContainer: {
        on: function(key, cb) {
            jest.spyOn(functionUtils, 'hideElement')
                .mockImplementation(() => { })
            cb()
        },
        _contentElement: [],
        _element: [{
            contains: function() {
                return true
            },
            react: function() {
            },
            removeChild: function() {
                return true
            }
        }],
        isHidden: true
    },
    confirmClose: function() {
    },
    loading: function() {
    },
    resize: function(cb) {
        cb()
    },
    receive: function() {
    },
    send: function() {
    },
    saveState: function() {
    },
    hideElement: function() {
    }
}
const props2 = {
    loadState: function() {
        return {}
    },
    glContainer: {
        on: function(key, cb) {
            jest.spyOn(functionUtils, 'hideElement')
                .mockImplementation(() => { })
            cb()
        },
        _contentElement: [],
        _element: [{
            contains: function() {
                return true
            },
            react: function() {
            },
            removeChild: function() {
                return true
            }
        }],
        isHidden: true
    },
    confirmClose: function() {
    },
    loading: function() {
    },
    resize: function(cb) {
        cb()
    },
    receive: function() {
    },
    send: function() {
    },
    saveState: function() {
    },
    hideElement: function() {
    }
}
const props3 = {
    loadState: function() {
        return {
            accountObj: {
                currency: 1,
                account_id: 1,
                account_name: 1,
                hin: 1,
                work_phone: 1,
                home_phone: 1,
                fax: 1,
                email: 1,
                address: 1,
                account_type: 1,
                advisor_code: 1
            }
        }
    },
    glContainer: {
        on: function(key, cb) {
            jest.spyOn(functionUtils, 'hideElement')
                .mockImplementation(() => { })
            cb()
        },
        _contentElement: [],
        _element: [{
            contains: function() {
                return true
            },
            react: function() {
            },
            removeChild: function() {
                return true
            }
        }],
        isHidden: true
    },
    confirmClose: function() {
    },
    loading: function() {
    },
    resize: function(cb) {
        cb()
    },
    receive: function() {
    },
    send: function() {
    },
    saveState: function() {
    },
    hideElement: function() {
    }
}
dataStorage.translate = () => { }
let x;
beforeEach(() => {
    x = new Compoment(props1);
});
test('AccountInfo', () => {
    const AccountInfo = render(<Compoment {...props1} />)
})

test('AccountInfo', () => {
    const AccountInfo = render(<Compoment {...props2} />)
})
test('AccountInfo', () => {
    const AccountInfo = render(<Compoment {...props3} />)
})

test('render', () => {
    x.state = null;
    const logger = jest.spyOn(log, 'log');
    x.render();
    expect(logger).toBeCalled();
})

test('handleResize', () => {
    x.dom = {
        className: 'khai',
        classList: {
            add: (a) => {
                x.dom.className = a;
            }
        }
    };
    x.handleResize(730);
    expect(x.dom.className).toEqual('single');
})
test('handleResize', () => {
    x.dom = {
        className: 'khai',
        classList: {
            add: (a) => {
                x.dom.className = a;
            }
        }
    };
    x.handleResize(1270);
    expect(x.dom.className).toEqual('half');
})
test('handleResize', () => {
    x.dom = {
        className: 'khai',
        classList: {
            add: (a) => {
                x.dom.className = a;
            }
        }
    };
    x.handleResize(9999);
    expect(x.dom.className).toEqual('lagre');
})

test('handleResize', () => {
    x.dom = false;
    x.handleResize(9999);
})

test('renderProducts', () => {
    x.dom = false;
    x.state.accountObj = { equity_trading: 1, warrants_trading: 1, options_trading: 1, future_trading: 1 }
    x.renderProducts();
    expect(x.renderProducts).toMatchSnapshot();
})

test('renderProducts', () => {
    x.dom = false;
    x.state.accountObj = { equity_trading: 1, warrants_trading: 1, options_trading: 1, future_trading: 1 }
    x.renderProducts();
    expect(x.renderProducts).toMatchSnapshot();
})

test('dataReceivedFromSearchAccount', () => {
    let data = true;
    x.changeAccount = jest.fn();
    x.state.accountObj = { equity_trading: 1, warrants_trading: 1, options_trading: 1, future_trading: 1 }
    x.dataReceivedFromSearchAccount(data);
    expect(x.changeAccount).toBeCalled();
})

test('dataReceivedFromSearchAccount', () => {
    let data = false;
    x.state.accountObj = { equity_trading: 1, warrants_trading: 1, options_trading: 1, future_trading: 1 }
    x.dataReceivedFromSearchAccount(data);
    x.changeAccount = jest.fn();
    expect(x.changeAccount).not.toBeCalled();
})

test('changeAccount', () => {
    let data = true;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: { account_id: 1 } };
        })
    x.state.accountObj = { equity_trading: 1, warrants_trading: 1, options_trading: 1, future_trading: 1 }
    props1.send = jest.fn();
    x.changeAccount(data, true);
    expect(props1.send).toBeCalled();
})

test('changeAccount', () => {
    let data = true;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: { account_id: 1 } };
        })
    x.state.accountObj = { equity_trading: 1, warrants_trading: 1, options_trading: 1, future_trading: 1 }
    props1.send = jest.fn();
    x.changeAccount(data, false);
    expect(props1.send).not.toBeCalled();
})

test('changeAccount', () => {
    let data = true;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: { khai: 1 } };
        })
    props1.saveState = jest.fn();
    x.changeAccount(data, false);
    expect(props1.saveState).not.toBeCalled();
})

test('changeAccount', () => {
    let data = true;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: {} };
        })
    props1.saveState = jest.fn();
    x.changeAccount(data, false);
    expect(props1.saveState).not.toBeCalled();
})

test('changeAccount', () => {
    let data = true;
    jest.spyOn(functionUtils, 'getSymbolAccountWhenFirstOpenLayout')
        .mockImplementation(() => {
            return { newAccount: {} };
        })
    props1.saveState = jest.fn();
    x.changeAccount(false, false);
    expect(props1.saveState).not.toBeCalled();
})
