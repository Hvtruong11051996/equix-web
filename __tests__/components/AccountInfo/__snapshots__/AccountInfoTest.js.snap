// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderProducts 1`] = `[Function]`;

exports[`renderProducts 2`] = `[Function]`;
